test/new-e2e/tests/agent-runtimes/check_disk/disk_test.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
+)
+
+type baseCheckSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+const metricCompareTolerance = 0.01
+
+func TestLinuxDiskSuite(t *testing.T) {
+	agentOptions := []agentparams.Option{}
+	suiteOptions := []e2e.SuiteOption{}
+	//TODO: remove once the test is ready for CI
+	if os.Getenv("CI_PIPELINE_ID") == "" {
+		// if running locally, use the hardcoded pipeline id
+		agentOptions = append(agentOptions,
+			// update pipeline id when you push changes to the disk check or the agent itself
+			agentparams.WithPipeline("61451328"),
+		)
+		// helpful for local runs
+		suiteOptions = append(suiteOptions, e2e.WithDevMode())
+	}
+
+	suiteOptions = append(suiteOptions, e2e.WithProvisioner(
+		awshost.Provisioner(
+			awshost.WithAgentOptions(
+				agentOptions...,
+			),
+		),
+	))
+
+	t.Parallel()
+	e2e.Run(t, &baseCheckSuite{},
+		suiteOptions...,
+	)
+}
+
+func (v *baseCheckSuite) TestCheckDisk() {
+	testCases := []struct {
+		name        string
+		checkConfig string
+		agentConfig string
+	}{
+		{
+			"default",
+			`init_config:
+instances:
+  - use_mount: false
+`,
+			``,
+		},
+	}
+
+	for _, testCase := range testCases {
+		v.Run(testCase.name, func() {
+			v.T().Log("run the disk check using old version")
+			pythonMetrics := v.runDiskCheck(testCase.agentConfig, testCase.checkConfig, false)
+			v.T().Log("run the disk check using new version")
+			goMetrics := v.runDiskCheck(testCase.agentConfig, testCase.checkConfig, true)
+
+			// assert the check output
+			ElementsMatch(v.T(), pythonMetrics, goMetrics, metricPayloadCompare)
+			v.T().Logf("the check emitted %d metrics", len(pythonMetrics))
+		})
+	}
+}
+
+type metricPayload struct {
+	name           string
+	tags           []string
+	points         []float64
+	ty             int32
+	unit           string
+	SourceTypeName string
+	interval       int64
+	originProduct  uint32
+	originCategory uint32
+	originService  uint32
+}
+
+func metricPayloadCompare(a, b metricPayload) bool {
+	return a.name == b.name &&
+		slices.Equal(a.tags, b.tags) &&
+		slices.EqualFunc(a.points, b.points, func(a, b float64) bool {
+			if a == 0 {
+				return b == 0
+			}
+			return math.Abs(a-b)/a <= metricCompareTolerance
+		}) &&
+		a.ty == b.ty &&
+		a.unit == b.unit &&
+		a.SourceTypeName == b.SourceTypeName &&
+		a.interval == b.interval &&
+		a.originProduct == b.originProduct &&
+		a.originCategory == b.originCategory &&
+		a.originService == b.originService
+}
+
+func (v *baseCheckSuite) runDiskCheck(agentConfig string, checkConfig string, useNewVersion bool) []metricPayload {
+	if useNewVersion {
+		agentConfig += "\nuse_diskv2_check: true"
+		checkConfig += "\n  loader: core"
+	}
+
+	v.UpdateEnv(awshost.Provisioner(
+		awshost.WithAgentOptions(
+			agentparams.WithAgentConfig(agentConfig),
+			agentparams.WithFile("/etc/datadog-agent/conf.d/conf.yaml", checkConfig, true),
```

Comment:
Should we write to `/etc/datadog-agent/conf.d/disk.d/conf.yaml` ?

Reply:
Ah yes, I'm not sure how that worked 😕